homeassistant/components/lifx/config_flow.py

```diff
@@ -1,16 +1,211 @@
+
+        return await self.async_step_confirm()
+
+    async def async_step_homekit(
+        self, discovery_info: zeroconf.ZeroconfServiceInfo
+    ) -> FlowResult:
+        """Handle a flow initialized by Homekit discovery."""
+        if self._async_in_progress() or self._async_current_entries():
+            return self.async_abort(reason="single_instance_allowed")
+
+        await self.async_set_unique_id(self._domain)
+        return await self.async_step_confirm()
+
+
+class LifxOptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle the LIFX Options Flow."""
+
+    def __init__(self, config_entry: config_entries.ConfigEntry) -> None:
+        """Initialize the LIFX options flow."""
+        self.config_entry = config_entry
+        self.updated_options: dict[str, Any] = {}
+
+    async def async_step_init(self, user_input=None):
+        """Handle configuration initiated by a user."""
+        return await self.async_step_discovery_options(user_input)
 
-from homeassistant.core import HomeAssistant
-from homeassistant.helpers import config_entry_flow
+    async def async_step_discovery_options(self, user_input=None):
+        """Handle custom discovery option dialog and store."""
+        errors = {}
+        current_options = self.config_entry.options
 
-from .const import DOMAIN
+        if user_input is not None:
+            self.updated_options = dict(current_options)
+            self.updated_options[CONF_DISCOVERY_INTERVAL] = user_input.get(
+                CONF_DISCOVERY_INTERVAL
+            )
+            self.updated_options[CONF_DUPLICATE_DISCOVERY] = user_input.get(
+                CONF_DUPLICATE_DISCOVERY
+            )
+            self.updated_options[CONF_MESSAGE_TIMEOUT] = user_input.get(
+                CONF_MESSAGE_TIMEOUT
+            )
+            self.updated_options[CONF_RETRY_COUNT] = user_input.get(CONF_RETRY_COUNT)
+            self.updated_options[CONF_GRACE_PERIOD] = user_input.get(CONF_GRACE_PERIOD)
+            self.hass.config_entries.async_update_entry(
+                self.config_entry, options=self.updated_options
+            )
 
+            reloaded = await self.hass.services.async_call(
+                "homeassistant",
+                "reload_config_entry",
+                service_data={"entry_id": self.config_entry.entry_id},
+                blocking=True,
+                limit=30,
+            )
+            title = (
+                "LIFX integration reloaded successfully."
+                if reloaded
+                else "LIFX integration could not be reloaded. Please restart Home Assistant."
+            )
 
-async def _async_has_devices(hass: HomeAssistant) -> bool:
-    """Return if there are devices that can be discovered."""
-    lifx_ip_addresses = await aiolifx.LifxScan(hass.loop).scan()
-    return len(lifx_ip_addresses) > 0
+            return self.async_create_entry(title=title, data=None)
 
+        options_schema = {
```

Comment:
IMHO, less is more; tweaking options like these should not be needed.
It is unclear to me why we would provide these options?

Reply:
The rationale is to account for local network configurations: busy wifi networks would want a longer discovery interval with a higher message timeout but a lower retry count, while folks who need faster discovery on smaller networks could tweak the discovery down, the timeout down and the retries up.

Essentially, this provides access to modify the values that are currently hard-coded to allow for better optimisation based on local conditions.

Reply:
Our integrations should be robust out of the box, that is not something a user of Home Assistant should be worried about or even need to tweak. Hence I generally don't agree with adding options like these.

If these options are needed; it means our default is not correct.

Reply:
The defaults are fine for about 90% of local networks. But there's no single set of options that would work for the more complex configurations involving VLANs, routing, mDNS reflecting, VPNs, etc. 

In that case, the user is already running a fairly custom network, for which they need custom configuration options.

I can make the options require the Advanced mode switch enabled?

Reply:
They're also a trade-off: I (personally) don't mind long discovery intervals because I never turn off my bulbs. But someone who does, would want way less than the default 60 seconds.

Reply:
I agree with @frenck, whilst this toggle resolves the issue introduced in 2022.05 (see https://github.com/home-assistant/core/issues/71585), requiring new users to find and understand this option is a poor experience.

Regarding the above mention of "VLANs, routing, mDNS reflecting, VPNs etc", I don't have any of these features enabled. I am running a simple single subnet network with no VLANs on the 192.168.1.x address space and was still encountering this issue.

Hope you don't mind if I suggest some alternate solutions which are seemless to the user:

- Will https://github.com/home-assistant/core/pull/71673 resolve this issue instead? Once it's deployed I'm happy to test.
- Alternatively the old code from 2022.04 worked - maybe we could revert to that while we investigate why 2022.05 introduced instability to some users

Love the work and contributions you've made @Djelibeybi and appreciate how responsive you have been. I am happy to test any changes using my setup :)

Reply:
Yes, #71673 fixes the issue of bulbs not being auto-rediscovered. It won't fix the duplicate discovery issue but that's generally a red herring anyway. I would prefer that PR get merged. T

This PR is a new feature that folks have requested in the past that I happened to be almost done with when the issue was opened.

Reply:
Note that you _are_ testing that fix if you install the beta custom component from https://github.com/Djelibeybi/ha-lifx-beta even if you don't touch any of the settings, as it has the fix included.